### PR TITLE
Malloc to Calloc calls for 0 initialize padding bytes

### PR DESCRIPTION
**Pull request by ttins**

### CODE REVIEW

**Comment by ttins** [comment]
Do not merge until [https://github.com/Azure/azure-iot-sdk-c/pull/1455] has been merged

File: adapters/condition_win32.c [hunk 1]
```diff
@@ -20,12 +20,12 @@ CONDITION;
 COND_HANDLE Condition_Init(void)
 {
     // Codes_SRS_CONDITION_18_002: [ Condition_Init shall create and return a CONDITION_HANDLE ]
-
     CONDITION* cond = (CONDITION*)malloc(sizeof(CONDITION));
 
     // Codes_SRS_CONDITION_18_008: [ Condition_Init shall return NULL if it fails to allocate the CONDITION_HANDLE ]
     if (cond != NULL)
     {
+        (void)memset(cond, 0, sizeof(CONDITION));
```
Comment (danewalton) [inline adapters/condition_win32.c]: nit: I prefer doing `sizeof({var_name})`. In this case:
```C
(void)memset(cond, 0, sizeof(cond));
```
Reason being if the type of `cond` were to ever change, the `sizeof()` call gets resolved correctly and you don't get out of sync.

Comment (ttins) [reply adapters/condition_win32.c]: I prefer to have that one calloced, but based on the UT and how it is mocking the memory management functions and redefining them 3 times it was currently out of our scope as it gets passed into umock as a mock function it seems to default to using malloc where it should be calling calloc so something is amiss in how umock is mocking it

Comment (danewalton) [reply adapters/condition_win32.c]: no i meant specifically the use of `sizeof()` not memset

**Pull request merged**